addons/mail/controllers/discuss/channel.py

```diff
@@ -61,6 +69,15 @@ def _process_request_for_all(self, store: Store, name, params):
                 [("channel_id", "=", params["channel_id"]), ("is_self", "=", True)],
             ):
                 member.unpin_dt = False if params["pinned"] else fields.Datetime.now()
+            if (params["request_has_unpinned_channels"]):
```

Comment:
I would not bother with this param, changing the pin value is not a frequent operation and computing `has_unpinned_channels` should not be so costly that we want to avoid it so much.

Always sending the value when doing the operation should be good enough.